packages/machine-extractor/src/extractAction.ts

```diff
@@ -411,10 +346,23 @@ function extractEventObject(
       }
     }
   });
+  return extracted;
+}
+
+// TODO: enforce return type to ActionEvent so `type` property is required
```

Comment:
@Andarist We don't yet have a validation in place, do we?  So I think this only enforces having `type` property in the type-level and is fine?

Reply:
hm, ye - the whole thing is kinda the "best effort" thing and there is no much validation in the whole thing. I think though that this shouldn't exactly be validated here.

We **want** to handle invalid things. I can edit (and fix) `sendTo({})` in the IDE and I should do the same with it in the Studio. So the goal is to transfer such a thing successfully to the Studio and validate it there + offer a way to correct the mistake